### PR TITLE
replace is-absolute-href with is-http-href

### PR DESCRIPTION
**Pull request by OsmondJiang**

### CODE REVIEW

File: src/docfx/restore/RestoreMap.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ public string GetUrlRestorePath(string docsetPath, string path)
         {
             Debug.Assert(!string.IsNullOrEmpty(path));
 
-            if (!HrefUtility.IsAbsoluteHref(path))
+            if (!HrefUtility.IsHttpHref(path))
```
Comment (yufeih) [inline src/docfx/restore/RestoreMap.cs]: nit: IsHttp?

File: src/docfx/lib/HrefUtility.cs [hunk 2]
```diff
@@ -66,6 +66,10 @@ public static bool IsAbsoluteHref(string str)
                 || Uri.TryCreate(str, UriKind.Absolute, out _);
         }
 
+        public static bool IsHttpHref(string str)
+            => !string.IsNullOrEmpty(str) && Uri.TryCreate(str, UriKind.Absolute, out var uriResult)
```
Comment (yufeih) [inline src/docfx/lib/HrefUtility.cs]: nit: `Startswith("http://")  || StartsWith("https://")` ?

is "https://///a.com" considered http?

File: src/docfx/restore/RestoreMap.cs (hunk 1 above)
Comment (OsmondJiang) [reply src/docfx/restore/RestoreMap.cs]: consistent with IsAbsoluteHref?

---
In reply to: [203931119](https://github.com/dotnet/docfx/pull/3105#discussion_r203931119) [](ancestors = 203931119)

File: src/docfx/lib/HrefUtility.cs (hunk 2 above)
Comment (OsmondJiang) [reply src/docfx/lib/HrefUtility.cs]: StartWith will not work for your case. https://///a.com is not http

---
In reply to: [203931273](https://github.com/dotnet/docfx/pull/3105#discussion_r203931273) [](ancestors = 203931273)

Comment (OsmondJiang) [reply src/docfx/lib/HrefUtility.cs]: will add test for your case

---
In reply to: [203931759](https://github.com/dotnet/docfx/pull/3105#discussion_r203931759) [](ancestors = 203931759,203931273)

**Pull request merged**